src/AspNetCore/Inc/applicationmanager.h

```diff
@@ -79,18 +79,18 @@ class APPLICATION_MANAGER
 
     ~APPLICATION_MANAGER()
     {
+        if (m_pFileWatcher != NULL)
```

Comment:
Why was this moved up?

Reply:
Stop file watcher first is safer trying to avoid the race that FCN comes to a shutting down applicaiton